app/services/support/filter.rb

```diff
@@ -19,14 +19,25 @@ def call
 
     attr_reader :model_data_scope, :filters
 
+    def provider_and_course_search(model_data_scope, filters)
+      search_params = { provider_name_or_code: filters[:provider_search], course_code: filters[:course_search] }
+      return model_data_scope if filters.values.none?
+
+      model_data_scope.provider_and_course_search(search_params)
+    end
+
     def text_search(model_data_scope, text_search)
       return model_data_scope if text_search.blank?
 
       model_data_scope.search(text_search)
```

Comment:
```suggestion
      model_data_scope.provider_and_course_search(provider_name_or_code: text_search)
```

It's looking like the original `#search` and `#provider_and_course_search` is now inherently same.
Maybe rename the original `#search` to `#provider_search`

And rename `#provider_and_course_search`  to `#search`